DotProduct.lua

```diff
@@ -26,6 +26,11 @@ function DotProduct:updateGradInput(input, gradOutput)
    local v2 = input[2]
    local not_batch = false
 
+   if #self.gradInput ~= 2 then
```

Comment:
Same comment as the one from `CosineDistance`. Also, the [buffer](https://github.com/szagoruyko/nn/blob/clearState/DotProduct.lua#L16) should also be cleaned